driver-sync/src/test/functional/com/mongodb/client/unified/ValueMatcher.java

```diff
@@ -39,92 +41,105 @@ final class ValueMatcher {
     }
 
     public void assertValuesMatch(final BsonValue expected, final BsonValue actual) {
-        assertValuesMatch(expected, actual, true, null, -1);
+        assertValuesMatch(expected, actual, null, -1);
     }
 
-    private void assertValuesMatch(final BsonValue initialExpected, final BsonValue actual, final boolean isRoot,
```

Comment:
It turns out that `actual` can be null in some cases.

Also,`isRoot` was not used so I killed it.